src/funcs.c

```diff
@@ -49,6 +49,9 @@
 #endif
 #include <exec/memory.h>
 #include <exec/types.h>
+#if defined(__amigaos4__)
+#define ASL_PRE_V38_NAMES
```

Comment:
Can you please elaborate about this line? Why is this needed?

Reply:
With AmigaOS 4.0 and onward, ASL requester structures have had pretty much all their members renamed. However, the old names are still available when `ASL_PRE_V38_NAMES` is defined ahead of including libraries/asl.h. This takes care of another compilation problem I was having with your branch, without needing another define set in `$CFLAGS_AOS4` in the makefiles -- since this is the only place where libraries/asl.h is included, it made sense to me to only define `ASL_PRE_V38_NAMES` alongside it.